tests/mnpe_test.py

```diff
@@ -151,87 +152,23 @@ def test_mnpe_api(flow_model: str, z_score_theta: str, use_embed_net: bool):
         posterior.sample((1,))
 
 
-class MixedSimulator(torch.nn.Module):
-    def __init__(self, sigma2):
-        super().__init__()
-        self.sigma2 = sigma2
-
-    def forward(self, theta):
-        # return x~N(mu_z, tau^2)
-        mu = theta[:, :2]
-        z = theta[:, -1]
-        # Select mu based on z
-        mu_z = torch.where(z == 0, mu[:, 0], mu[:, 1])
-        # Sample from N(mu_z, sigma2)
-        x = Normal(loc=mu_z, scale=torch.sqrt(self.sigma2)).rsample()
-        return x.reshape(-1, 1)
-
-
-@pytest.mark.slow
-@pytest.mark.parametrize("flow_model", ("nsf", "zuko_nsf"))
-def test_mnpe_accuracy(
-    flow_model: str,
-):
-    """Test MNPE accuracy."""
-    num_simulations = 1000
-    num_posterior_samples = 1000
-
-    z_score_theta = "none"
-    use_embed_net = False
-
-    # Prio
-    mu_0 = torch.tensor([-1.0])
-    mu_1 = torch.tensor([1.0])
-    tau2 = torch.tensor([0.1])
+def reference_posterior_mog(tau2, sigma2, mu_0, mu_1, x_o, q, num_posterior_samples):
+    """Reference posterior for a mixture of two Gaussians.
```

Comment:
to avoid confusion with a classical (continuous MoG) I would call it a Bernoulli-Gaussian mixture